local_planner/src/nodes/planner_functions.cpp

```diff
@@ -45,13 +59,36 @@ void filterPointCloud(
     }
   }
 
-  cropped_cloud.header.stamp = complete_cloud[0].header.stamp;
-  cropped_cloud.header.frame_id = complete_cloud[0].header.frame_id;
-  cropped_cloud.height = 1;
-  cropped_cloud.width = cropped_cloud.points.size();
-  if (cropped_cloud.points.size() <= min_cloud_size) {
-    cropped_cloud.points.clear();
-    cropped_cloud.width = 0;
+  // combine with old cloud
+  for (const pcl::PointXYZI& xyzi : old_cloud) {
+    // adding older points if not expired and space is free according to new
+    // cloud
+    if (histogram_box.isPointWithinBox(xyzi.x, xyzi.y, xyzi.z)) {
+      distance = (position - toEigen(xyzi)).norm();
+      if (distance < histogram_box.radius_) {
+        PolarPoint p_pol = cartesianToPolar(toEigen(xyzi), position);
+        Eigen::Vector2i p_ind = polarToHistogramIndex(p_pol, ALPHA_RES / 2);
+        if (high_res_histogram.get_age(p_ind.y(), p_ind.x()) == 0 &&
```

Comment:
You are now using the "age" layer of the histogram for something completely new - it's not containing the age but the binary information whether we have a point in this cell yet. This is fine, but you should rename it in a meaningful way in order to keep things maintainable and understandable down the road

Reply:
Maybe better would be to keep NaN in the distance matrix if there is no sample, rather than a separate layer.

Reply:
removed age layer from histogram class, as age is now stored in the pointcloud per point. Stuck to using 0 for empty cells instead of NANs, as we add distances to bins and divide by n points. This would get much more complicated it empty cells were NAN